### ISSUE TITLE
Inference problem with text-to-speech (custom model)

### ISSUE THREAD

**Issue by kormoczi** [espnet/espnet#3646]
Hi,

I started to play with the egs2/css10/tts1 recipe, to create a text-to-speech model.
I went through the steps from 1 to 7 (mostly with the default settings, just with fewer number of epochs at the beginning).
The results from step 7 (inference) looks nice (not the best quality yet, of course, but quite good).

But when I try to do the inference from my own python app, the result is bad (the wav output contains just one or two phonemes, nothing valuable). This code can work well with pretrained models...

What am I doing wrong? Any advice?
Thanks!

```
import torch
import soundfile
from espnet2.bin.tts_inference import Text2Speech
from parallel_wavegan.utils import load_model
import subprocess
import kaldiio

MODEL_FILE = "/root/espnet/egs2/css10/tts1/exp/tts_train_raw_phn_none/latest.pth"
CONFIG_FILE = "/root/espnet/egs2/css10/tts1/exp/tts_train_raw_phn_none/config.yaml"
SPEAKER_FILE = "/root/espnet/egs2/css10/tts1/dump/xvector/dev_phn/spk_xvector.ark"

text2speech = Text2Speech(
    train_config=CONFIG_FILE,
    model_file=MODEL_FILE,
    device="cuda",
)

xvector_ark = SPEAKER_FILE
xvectors = {k: v for k, v in kaldiio.load_ark(xvector_ark)}
spks = list(xvectors.keys())
spk = spks[0]
spembs = xvectors[spk]
speech = torch.randn(50000,) * 0.01

text = "hello"
wav = text2speech(text, speech=speech, spembs=spembs)["wav"]

with open('tts_test.wav', 'wb') as wav_file:
    soundfile.write(wav_file, wav.cpu().numpy(), text2speech.fs, format="wav")

```

**Comment by Fhrozen** [comment]
It will be better if u use `from_pretrained`
```
Text2Speech.from_pretrained(
            CONFIG_FILE,
            vocoder_tag=vocoder_tag,
            device="cuda",
            # Only for Tacotron 2
            threshold=0.5,
            minlenratio=0.0,
            maxlenratio=10.0,
            use_att_constraint=False,
            backward_window=1,
            forward_window=3,
            # Only for FastSpeech, FastSpeech2, & VITS 
            speed_control_alpha=1.0,
        )
```

**Comment by kan-bayashi** [comment]
Since css10 is multilingual, we perform the g2p conversion in data prep stage.
So the input is needed to be phoneme, not text.
And css10 does not include english.

**Comment by kan-bayashi** [comment]
> I have also tried to use the pack.py to create an archive from the model, but with a lot of different tries, no success, unfortunately (and cant find it in the documentation how to do it properly)...

Give me the command and log.

**Comment by kan-bayashi** [comment]
https://github.com/espnet/espnet/blob/66facd2859977d23349cf74d5c8f5394e5cb6784/egs2/TEMPLATE/tts1/tts.sh#L1075
You can pack usage here.

**Comment by kormoczi** [comment]
I went forward with the training... To my big surprise, around epoch no 60, the logs started to say, that "There are no improvements in this epoch" (num_items_per_epoch=500).
Is this normal?
I have downloaded only one language, and trained for that one only (modified the langs parameter in the local/data.sh accordingly).
Is it better to train with multiple languages, even if I am planning to use just one?
Or still valuable to go forward with the training, despite these logs?
Or do I need to play with some hyperparameters?
By the way, at the moment I can feel the following issues with the actual status of model - and that is why I think I need to train more:
1. after commas usually there is a very long pause/silence (I am not sure if I am doing the preprocessing - phoneme conversion right or not)
2. end of the line/sentence sometimes missing/cut off
3. intonation for the questioning and exclamation sentences not so clear (not much difference from declarative sentences)
4. rarely I have some bad phonemes (more often if the text contain multiple sentences)
